src/tests/blueprint/t_blueprint_mesh_verify.cpp

```diff
@@ -0,0 +1,1163 @@
+        n["elements"]["b"]["connectivity"].set(DataType::int32(1));
+        EXPECT_TRUE(blueprint::mesh::topology::unstructured::verify(n,info));
+
+        n["elements"]["c"]["shape"].set("tri");
+        n["elements"]["c"]["connectivity"].set(DataType::int32(5));
+        EXPECT_TRUE(blueprint::mesh::topology::unstructured::verify(n,info));
+    }
+
+    { // Multiple Shape Topology Stream Tests //
+        // FIXME: Implement once multiple unstructured shape topologies are implemented.
+        n["elements"].reset();
+    }
+}
+
+
+//-----------------------------------------------------------------------------
+TEST(conduit_blueprint_mesh_verify, topology_types)
+{
+    Node n, info;
+
+    const std::string topology_types[] = {"uniform", "rectilinear", "structured", "unstructured"};
+    for(index_t ti = 0; ti < 4; ti++)
+    {
+        n.reset();
+        n.set(topology_types[ti]);
+        EXPECT_TRUE(blueprint::mesh::topology::type::verify(n,info));
+    }
+
+    n.reset();
+    n.set(0);
+    EXPECT_FALSE(blueprint::mesh::topology::type::verify(n,info));
+
+    n.reset();
+    n.set("explicit");
+    EXPECT_FALSE(blueprint::mesh::topology::type::verify(n,info));
+}
+
+
+//-----------------------------------------------------------------------------
+TEST(conduit_blueprint_mesh_verify, topology_shape)
+{
+    Node n, info;
+    EXPECT_FALSE(blueprint::mesh::topology::shape::verify(n,info));
+
+    const std::string shape_types[] = {"point", "line", "tri", "quad", "tet", "hex"};
+    for(index_t ti = 0; ti < 6; ti++)
+    {
+        n.reset();
+        n.set(shape_types[ti]);
+        EXPECT_TRUE(blueprint::mesh::topology::shape::verify(n,info));
+    }
+
+    n.reset();
+    n.set(10);
+    EXPECT_FALSE(blueprint::mesh::topology::shape::verify(n,info));
+
+    n.reset();
+    n.set("poly");
+    EXPECT_FALSE(blueprint::mesh::topology::shape::verify(n,info));
+}
+
+
+//-----------------------------------------------------------------------------
+TEST(conduit_blueprint_mesh_verify, topology_general)
+{
+    VerifyFun verify_topology_funs[] = {
+        blueprint::mesh::topology::verify,
+        verify_topology_protocol};
+
+    for(index_t fi = 0; fi < 2; fi++)
+    {
+        VerifyFun verify_topology = verify_topology_funs[fi];
+
+        Node mesh, info;
+        EXPECT_FALSE(verify_topology(mesh,info));
+
+        blueprint::mesh::examples::braid("quads",10,10,1,mesh);
+        Node& n = mesh["topologies"]["mesh"];
+
+        { // Type Field Tests //
+            n.remove("type");
+            EXPECT_FALSE(verify_topology(n,info));
+            n["type"].set("explicit");
+            EXPECT_FALSE(verify_topology(n,info));
+
+            // FIXME: Remove the comments from the following line once the verify functions
+            // for uniform and rectilinear topologies have been implemented.
+            const std::string topology_types[] = {/*"uniform", "rectilinear", */"structured"};
```

Comment:
These functions do exist and are currently implement as no-ops, but wiring them up here wouldn't be too helpful.  The purpose of this part of the test case is to verify that a valid "explicit" topology isn't accepted as any other form of topology when fed through the `blueprint::mesh::topology::verify` function (essentially just ensuring that the if clauses in this function point to the correct subroutines).  Since the "uniform" and "rectilinear" topology tests are no-ops that return true, calling them here would cause this test to fail (this is actually good because this indicates that these verify functions aren't returning values we'd expect).  Ultimately, I think that it's best to keep this commented out for now, especially considering that we already have no-op coverage for the "uniform" and "rectilinear" cases in the "topology_uniform" and "topology_rectilinear" test cases above.